infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/cache/jpa/JpaKubernetesRuntimeStateCache.java

```diff
@@ -93,8 +93,7 @@ public WorkspaceStatus getStatus(RuntimeIdentity id) throws InfrastructureExcept
       Optional<KubernetesRuntimeState> runtimeStateOpt = get(id);
 
       if (!runtimeStateOpt.isPresent()) {
-        throw new InfrastructureException(
-            "Runtime state for workspace with id '" + id.getWorkspaceId() + "' was not found");
+        return WorkspaceStatus.STOPPED;
```

Comment:
If an exception is not thrown here anymore it looks like STOPPED state will be saved as a runtime state here https://github.com/eclipse/che/blob/782db9c98ce146f3bfc32207fe3540c628047d91/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java#L627. WDYT?

Reply:
BTW should we change the javadocs of this method if we want to introduce this change?

Reply:
Nice catch!
This change was needed in my previous solution and is not needed in the current one. I guess it's better to revert it for time being. I'll take a look.

Reply:
Reverted